WordPress/src/test/java/org/wordpress/android/ui/posts/editor/EditorActionsProviderTest.kt

```diff
@@ -0,0 +1,172 @@
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.PUBLISH_NOW)
+    }
+
+    @Test
+    fun `Verify an error is logged when the user doesn't have publishing rights and works with a SCHEDULED post`() {
+        // Arrange & Act
+        val (_, _) =
+                getPrimaryAndSecondaryActions(PostStatus.SCHEDULED, userCanPublish = false)
+        // Assert
+        verify(crashLoggingUtilsWrapper, times(2)).log(any<String?>())
+    }
+
+    @Test
+    fun `Verify actions of a TRASHED post`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.TRASHED, userCanPublish = true)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.SAVE)
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.SAVE_AS_DRAFT)
+    }
+
+    @Test
+    fun `Verify an error is logged when the user doesn't have publishing rights and works with a TRASHED post`() {
+        // Arrange & Act
+        val (_, _) =
+                getPrimaryAndSecondaryActions(PostStatus.TRASHED, userCanPublish = false)
+        // Assert
+        verify(crashLoggingUtilsWrapper, times(2)).log(any<String?>())
+    }
+
+    @Test
+    fun `Verify actions of a PENDING post`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.PENDING, userCanPublish = true)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.SAVE)
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.PUBLISH_NOW)
+    }
+
+    @Test
+    fun `Verify actions of a PENDING post when the user doesn't have publishing rights`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.PENDING, userCanPublish = false)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.SUBMIT_FOR_REVIEW)
+        // TODO Would it make sense to have "SAVE/SAVE_AS_DRAFT" action here?
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.NONE)
+    }
+
+    @Test
+    fun `Verify actions of a PRIVATE post`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.PRIVATE, userCanPublish = true)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.UPDATE)
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.NONE)
+    }
+
+    @Test
+    fun `Verify an error is logged when the user doesn't have publishing rights and works with a PRIVATE post`() {
+        // Arrange & Act
+        val (_, _) =
+                getPrimaryAndSecondaryActions(PostStatus.PRIVATE, userCanPublish = false)
+        // Assert
+        verify(crashLoggingUtilsWrapper, times(2)).log(any<String?>())
+    }
+
+    @Test
+    fun `Verify actions of a UNKNOWN post`() {
```

Comment:
English nitpick 😅 unknown is normally preceded by "an" 😄

Reply:
❤️ Finally someone who is willing to correct my mistakes in English - I honestly appreciate it;)!
Fixed in 19492ed4c6166f9254f03c4b6a2d79691ca70ff3

Reply:
😄